examples/MomentoRedisLambdaExample/MomentoRedisExampleLambdaHandler/src/MomentoRedisExampleLambdaHandler/RedisCache.cs

```diff
@@ -0,0 +1,143 @@
+        Console.WriteLine($"\tIssued redis set request for {key}");
+        var setResponse = await setResponseTask;
+        Console.WriteLine($"\tDone awaiting redis set response for {key}");
+        stopWatch.Stop();
+        var duration = stopWatch.ElapsedMilliseconds;
+        // Console.WriteLine($"\tReturning redis set response for {key}: {setResponse}");        
+        Console.WriteLine($"\tReturning redis set response for {key}");        
+        return new RedisSetResult(Response: setResponse, ElapsedMillis: duration);
+    }
+    
+#pragma warning disable CS1998
+    public async Task<List<Task<RedisGetResult>>> DoRedisGets(int numRedisGetRequests, bool compressValue)
+#pragma warning restore CS1998
+    {
+        Console.WriteLine("Executing Redis Gets");
+        List<Task<RedisGetResult>> redisGetTasks = new List<Task<RedisGetResult>>();
+
+        for (int i = 1; i <= numRedisGetRequests; i++)
+        {
+            Console.WriteLine("Executing RedisGets");
+            string key = $"key-{i}";
+            redisGetTasks.Add(ExecuteGet(key, compressValue));
+        }
+
+        return redisGetTasks;
+    }
+    
+    private async Task<RedisGetResult> ExecuteGet(string key, bool compressValue)
+    {
+        
+        Console.WriteLine($"Beginning redis get request for {key}");
+        var withCompressionStopWatch = new Stopwatch();
+        var stopWatch = Stopwatch.StartNew();
+        if (compressValue)
+        {
+            withCompressionStopWatch = Stopwatch.StartNew();
+        }
+        var getResponseTask = _db.StringGetAsync(key);
+        Console.WriteLine($"\tIssued redis get request for {key}");
+        var getResponse = await getResponseTask;
+        Console.WriteLine($"\tDone awaiting redis get response for {key}");
+        stopWatch.Stop();
+        var responseElapsedMillis = stopWatch.ElapsedMilliseconds;
+        long withDecompressionElapsedMillis = 0;
+
+        if (compressValue)
+        {
+            if (getResponse.HasValue)
+            {
+                var decompressed = Compression.Decompress(getResponse!);
+                Console.WriteLine($"\tResponse finished decompressing for {key}");
+                withCompressionStopWatch.Stop();
+                withDecompressionElapsedMillis = withCompressionStopWatch.ElapsedMilliseconds;
+            }
+            else
+            {
+                Console.WriteLine($"\tValue not found for {key}");
+            }
+        }
+        
+        var getResult = new RedisGetResult(
+            Response: getResponse,
+            ElapsedMillis: responseElapsedMillis,
+            ElapsedMillisWithDecompression: withDecompressionElapsedMillis
+        );
+        // Console.WriteLine($"\tReturning get response for {key}: {getResult}");
+        Console.WriteLine($"\tReturning get response for {key}");
+        return getResult;
+    }
+
+    public async void LogRedisSetResult(List<Task<RedisSetResult>> redisSetTasks)
```

Comment:
same comment on this function and the next one about realizing the list of tasks prior to calling this and making it sync.